teuthology/openstack/__init__.py

```diff
@@ -923,40 +959,49 @@ def get_user_data(self):
         log.debug("get_user_data: " + content + " written to " + path)
         return path
 
+    def key_pair(self):
```

Comment:
Whats the reason for having three methods that do the same thing?

Reply:
They do not do the same, each of which is a getter which returns pattern based on the name, it is a coincidence that they have the same pattern. But in general, this is a kind of refactoring to separate entities. Nevertheless, all of them have the same value and type, they are different in nature: server name, server security group, worker security group, and referring to the corresponding symbol is self-documented.